# **Project Description: Descriptive Analysis of Customer Purchase Patterns**
Project Title: Understanding Customer Purchase Patterns at XYZ Retail
Objective: The primary goal of this project is to conduct a descriptive analysis of customer purchase data at XYZ Retail. Through this analysis, we aim to summarize key characteristics of customer purchases, identify trends, and generate insights that can inform marketing strategies and inventory management.


In [5]:
!pip install dash

import dash
from dash import dcc, html
import plotly.express as px
import pandas as pd
import numpy as np
from datetime import datetime, timedelta

# --- Generate Synthetic Data ---
np.random.seed(42)

# Define constants
num_rows = 1000
start_date = datetime(2023, 1, 1)
end_date = datetime(2024, 1, 1)

# Generate random dates within the given range
date_range = [start_date + timedelta(days=np.random.randint(0, (end_date - start_date).days)) for _ in range(num_rows)]

# Define categories and payment methods
product_categories = ['Electronics', 'Clothing', 'Groceries', 'Furniture', 'Toys']
payment_methods = ['Credit Card', 'Cash', 'Digital Payment']

# Generate dataset
data = {
    'Transaction ID': range(1, num_rows + 1),
    'Customer ID': np.random.randint(1000, 5000, size=num_rows),
    'Purchase Date': date_range,
    'Product Category': np.random.choice(product_categories, size=num_rows),
    'Quantity': np.random.randint(1, 5, size=num_rows),
    'Price': np.round(np.random.uniform(5, 500, size=num_rows), 2),
    'Payment Method': np.random.choice(payment_methods, size=num_rows),
    'Location': np.random.choice(['New York', 'Los Angeles', 'Chicago', 'Houston', 'Miami'], size=num_rows)
}

df = pd.DataFrame(data)

# Convert date to datetime format
df['Purchase Date'] = pd.to_datetime(df['Purchase Date'])

# --- Data Aggregation for Visualization ---
# 1. Sales Trend Over Time
transactions_per_month = df.set_index('Purchase Date').resample('M').size()
sales_trend_fig = px.line(x=transactions_per_month.index, y=transactions_per_month.values,
                          labels={'x': 'Month', 'y': 'Number of Transactions'},
                          title="Sales Trend Over Time")

# 2. Product Category Distribution (Top 5)
category_distribution = df['Product Category'].value_counts().nlargest(5)
category_fig = px.bar(x=category_distribution.values, y=category_distribution.index,
                      orientation='h', labels={'x': 'Number of Purchases', 'y': 'Product Category'},
                      title="Product Category Distribution")

# 3. Payment Method Distribution
payment_counts = df['Payment Method'].value_counts()
payment_fig = px.pie(names=payment_counts.index, values=payment_counts.values,
                     title="Payment Method Distribution")

# --- Build Dash App ---
app = dash.Dash(__name__)

app.layout = html.Div(children=[
    html.H1("Customer Purchase Analysis Dashboard", style={'text-align': 'center'}),

    html.Div(children=[
        dcc.Graph(figure=sales_trend_fig, style={'width': '33%', 'display': 'inline-block'}),
        dcc.Graph(figure=category_fig, style={'width': '33%', 'display': 'inline-block'}),
        dcc.Graph(figure=payment_fig, style={'width': '33%', 'display': 'inline-block'}),
    ]),
])

# Run the app
if __name__ == '__main__':
   app.run(debug=True)

<ipython-input-5-b5112a027e9f>:44: FutureWarning:

'M' is deprecated and will be removed in a future version, please use 'ME' instead.



<IPython.core.display.Javascript object>

In [6]:
# 1. Sales Trend Insights
print("\n--- Sales Trend Over Time ---")
print(transactions_per_month.describe())  # Summary statistics for monthly transactions
print("\nPeak Sales Month:", transactions_per_month.idxmax().strftime('%B %Y'),
      "with", transactions_per_month.max(), "transactions.")

# 2. Product Category Distribution Insights
print("\n--- Top 5 Product Categories ---")
print(category_distribution)
print("\nMost Purchased Product Category:", category_distribution.idxmax(),
      "with", category_distribution.max(), "purchases.")

# 3. Payment Method Distribution Insights
print("\n--- Payment Method Distribution ---")
print(payment_counts)
print("\nMost Used Payment Method:", payment_counts.idxmax(),
      "with", payment_counts.max(), "transactions.")


--- Sales Trend Over Time ---
count     12.000000
mean      83.333333
std       11.734435
min       60.000000
25%       77.250000
50%       83.000000
75%       90.250000
max      104.000000
dtype: float64

Peak Sales Month: August 2023 with 104 transactions.

--- Top 5 Product Categories ---
Product Category
Furniture      226
Toys           209
Clothing       203
Electronics    193
Groceries      169
Name: count, dtype: int64

Most Purchased Product Category: Furniture with 226 purchases.

--- Payment Method Distribution ---
Payment Method
Digital Payment    344
Cash               335
Credit Card        321
Name: count, dtype: int64

Most Used Payment Method: Digital Payment with 344 transactions.


# **Insights and Findings**
The descriptive analysis of customer purchase patterns at XYZ Retail reveals several key insights. The peak sales period occurred in August 2023, with the highest number of transactions (104), suggesting a potential seasonal demand, possibly influenced by back-to-school shopping or summer promotions. The top-selling product category was Furniture (226 purchases), followed by Toys (209), Clothing (203), Electronics (193), and Groceries (169). This indicates a strong demand for home-related products and children’s items, which may correlate with seasonal promotions or consumer preferences.

In terms of payment preferences, Digital Payments (344 transactions) were the most frequently used method, surpassing Cash (335 transactions) and Credit Cards (321 transactions). This suggests a growing trend toward digital transactions, which could be attributed to increasing convenience, mobile payment adoption, or store incentives for cashless payments.

# **Recommendations**
Inventory Management:

Given that Furniture is the most purchased category, XYZ Retail should ensure adequate stock levels for high-demand furniture items, especially leading up to peak shopping months.

Since Toys and Clothing also have high purchase rates, targeted promotions and bulk stocking should be considered during peak seasons.

Targeted Marketing Strategies:

The spike in transactions in August suggests a seasonal shopping trend. Marketing campaigns could be optimized around this period by offering special discounts or bundle deals on high-demand categories.

Digital payments are gaining popularity, so offering exclusive discounts for mobile or online transactions could further drive engagement.

Promotional Strategies:

Loyalty programs or cashback rewards for digital payment users can help enhance customer retention and drive higher transaction values.

Since Furniture and Toys are top-selling categories, cross-promotional strategies such as “Buy Furniture, Get a Discount on Home Decor” could increase sales volume.

By leveraging these insights, XYZ Retail can enhance its inventory planning, marketing effectiveness, and customer experience, ultimately driving increased sales and customer satisfaction.